gapis/replay/gpu_profile.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package replay
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/service"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+// GpuProfile replays the trace and writes a Perfetto trace of the replay
+func GpuProfile(ctx context.Context, capturePath *path.Capture, device *path.Device, opts *service.TraceOptions) (*service.GpuProfileResponse, error) {
+	c, err := capture.ResolveGraphicsFromPath(ctx, capturePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if device != nil {
```

Comment:
if we're checking `device != nil ` then maybe we also need to deal with the `else` condition.